src/lib/commands/drivers-util.ts

```diff
@@ -33,20 +34,56 @@ export const listDrivers = async (client: SmartThingsClient, includeAllOrganizat
 		? forAllOrganizations(client, orgClient => orgClient.drivers.list())
 		: client.drivers.list()
 
-
+/**
+ * Filter the driver currently in use by a device out of a list of drivers.
+ */
+export const withoutCurrentDriver = async (client: SmartThingsClient, deviceId: string, drivers: DriverChoice[]): Promise<DriverChoice[]> => {
+	const device = await client.devices.get(deviceId)
+	const currentDriverId = device.lan?.driverId ??
+		device.matter?.driverId ??
+		device.zigbee?.driverId ??
+		device.zwave?.driverId
+
+	return drivers.filter(driver => driver.driverId !== currentDriverId)
+}
+export const listAllAvailableDrivers = async (client: SmartThingsClient, edgeClient: EdgeClient, deviceId: string, hubId: string): Promise<DriverChoice[]> => {
+	const installedDrivers = await edgeClient.hubs.listInstalled(hubId)
+	const defaultDrivers = (await client.drivers.listDefault())
+		.filter(driver => !installedDrivers.find(installed => installed.driverId === driver.driverId))
+	return withoutCurrentDriver(client, deviceId, [...installedDrivers, ...defaultDrivers ])
+}
+export const listMatchingDrivers = async (client: SmartThingsClient, edgeClient: EdgeClient, deviceId: string, hubId: string): Promise<DriverChoice[]> =>
+	withoutCurrentDriver(client, deviceId, await edgeClient.hubs.listInstalled(hubId, deviceId))
+
+/**
+ * When presenting a list of drivers to choose from, we only use the `driverId` and `name` fields.
+ * Using this type instead of `EdgeDriverSummary` allows the caller of `chooseDriver` (below)
+ * to use functions that return other objects as long as they include these two fields.
+ */
+export type DriverChoice = Pick<EdgeDriverSummary, 'driverId' | 'name'>
+export interface ChooseDriverOptions extends ChooseOptions {
+	/**
+	 * By default drivers owned by the user are included, using `command.client.drivers.list()`
+	 * but if you need a different list of drivers, you can include your own function here.
+	 */
+	listItems: () => Promise<DriverChoice[]>
+}
```

Comment:
Nit: it would be nice to have spaces between each export in this file.

Reply:
I added the spaces. I had intentionally put them together because they are so closely related to each other but I like both ways.